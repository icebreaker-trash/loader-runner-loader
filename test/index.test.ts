import path from 'node:path'
import fs from 'node:fs'
import {
  getMemfsCompiler5,
  compile,
  readAssets,
  createLoader,
  getErrors,
  getWarnings
} from 'webpack-build-utils'
import MiniCssExtractPlugin from 'mini-css-extract-plugin'
// import { runLoaders } from 'promisify-loader-runner'
// 测试目录下有 wxml，会导致模块找不到！！！！
// 同时document.querySelector('#app').innerHTML = 要有
// 太奇怪了！！

// 本质是一个 postcss 引用问题，因为 bg-[url(./xxx)] 会导致 webpack 去加载它，这才是真正的原因！！！
describe('[Default]', () => {
  it('multipleContexts', async () => {
    // process.chdir(process.cwd())
    const context = path.resolve(__dirname, './fixtures/multiple-contexts')
    const customCompiler = getMemfsCompiler5({
      mode: 'production',

      optimization: {
        sideEffects: false
      },
      entry: {
        index: './src/index.js',
        module: './src/module/index.js'
      },
      context,
      plugins: [new MiniCssExtractPlugin()],
      //
      module: {
        rules: [
          {
            test: /\.css$/i,
            use: [MiniCssExtractPlugin.loader, 'css-loader', 'postcss-loader']
          }
        ]
      }
    })
    // @ts-ignore
    // customCompiler.outputFileSystem = fs
    const stats = await compile(customCompiler)
    const assets = readAssets(customCompiler, stats)
    expect(assets).toMatchSnapshot('assets')
    expect(getErrors(stats)).toMatchSnapshot('errors')
    expect(getWarnings(stats)).toMatchSnapshot('warnings')
  })
})
