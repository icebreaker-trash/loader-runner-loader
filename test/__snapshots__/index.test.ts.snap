// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[Default] > loader-runner-loader case 1 > assets 1`] = `
{
  "index.css": "

/* .a {
  color: red;
} */
.after\\\\:content-\\\\[\\\\'index\\\\.js\\\\'\\\\]::after {
    --tw-content: 'index.js';
    content: var(--tw-content)
}
.after\\\\:content-\\\\[\\\\'index\\\\.wxml\\\\'\\\\]::after {
    --tw-content: 'index.wxml';
    content: var(--tw-content)
}

",
  "index.js": "(()=>{\\"use strict\\";document.querySelector(\\"#app\\").innerHTML='\\\\n<main class=\\"after:content-[\\\\'index.js\\\\']\\"\\">\\\\n</main>\\\\n'})();",
}
`;

exports[`[Default] > loader-runner-loader case 1 > errors 1`] = `[]`;

exports[`[Default] > loader-runner-loader case 1 > warnings 1`] = `[]`;

exports[`[Default] > multipleContexts > assets 1`] = `
{
  "index.css": "

/* .a {
  color: red;
} */
.after\\\\:content-\\\\[\\\\'index\\\\.js\\\\'\\\\]::after {
    --tw-content: 'index.js';
    content: var(--tw-content)
}
.after\\\\:content-\\\\[\\\\'index\\\\.wxml\\\\'\\\\]::after {
    --tw-content: 'index.wxml';
    content: var(--tw-content)
}

",
  "index.js": "(()=>{\\"use strict\\";document.querySelector(\\"#app\\").innerHTML='\\\\n<main class=\\"after:content-[\\\\'index.js\\\\']\\"\\">\\\\n</main>\\\\n'})();",
  "module.css": "

/* .b {
  color: blue;
} */
.after\\\\:content-\\\\[\\\\'index\\\\.js\\\\'\\\\]::after {
    --tw-content: 'index.js';
    content: var(--tw-content)
}
.after\\\\:content-\\\\[\\\\'index\\\\.wxml\\\\'\\\\]::after {
    --tw-content: 'index.wxml';
    content: var(--tw-content)
}

",
  "module.js": "(()=>{\\"use strict\\";document.querySelector(\\"#app\\").innerHTML=\\"\\\\n<main class=\\\\\\"after:content-['m-index.js']\\\\\\">\\\\n</main>\\\\n\\"})();",
}
`;

exports[`[Default] > multipleContexts > errors 1`] = `[]`;

exports[`[Default] > multipleContexts > warnings 1`] = `[]`;

exports[`[Default] > multipleContexts case 0 > assets 1`] = `
{
  "index.css": "

/* .a {
  color: red;
} */
.after\\\\:content-\\\\[\\\\'index\\\\.js\\\\'\\\\]::after {
    --tw-content: 'index.js';
    content: var(--tw-content)
}
.after\\\\:content-\\\\[\\\\'index\\\\.wxml\\\\'\\\\]::after {
    --tw-content: 'index.wxml';
    content: var(--tw-content)
}

",
  "index.js": "(()=>{\\"use strict\\";document.querySelector(\\"#app\\").innerHTML='\\\\n<main class=\\"after:content-[\\\\'index.js\\\\']\\"\\">\\\\n</main>\\\\n'})();",
  "module.css": "

/* .b {
  color: blue;
} */
.after\\\\:content-\\\\[\\\\'index\\\\.js\\\\'\\\\]::after {
    --tw-content: 'index.js';
    content: var(--tw-content)
}
.after\\\\:content-\\\\[\\\\'index\\\\.wxml\\\\'\\\\]::after {
    --tw-content: 'index.wxml';
    content: var(--tw-content)
}

",
  "module.js": "(()=>{\\"use strict\\";document.querySelector(\\"#app\\").innerHTML=\\"\\\\n<main class=\\\\\\"after:content-['m-index.js']\\\\\\">\\\\n</main>\\\\n\\"})();",
}
`;

exports[`[Default] > multipleContexts case 0 > errors 1`] = `[]`;

exports[`[Default] > multipleContexts case 0 > warnings 1`] = `[]`;

exports[`[Default] > multipleContexts case 1 > assets 1`] = `
{
  "index.css": "

/* .a {
  color: red;
} */
.after\\\\:content-\\\\[\\\\'index\\\\.js\\\\'\\\\]::after {
    --tw-content: 'index.js';
    content: var(--tw-content)
}
.after\\\\:content-\\\\[\\\\'index\\\\.wxml\\\\'\\\\]::after {
    --tw-content: 'index.wxml';
    content: var(--tw-content)
}

",
  "index.js": "(()=>{\\"use strict\\";document.querySelector(\\"#app\\").innerHTML='\\\\n<main class=\\"after:content-[\\\\'index.js\\\\']\\"\\">\\\\n</main>\\\\n'})();",
  "module.css": "

/* .b {
  color: blue;
} */
.after\\\\:content-\\\\[\\\\'m-index\\\\.js\\\\'\\\\]::after {
    --tw-content: 'm-index.js';
    content: var(--tw-content)
}
.after\\\\:content-\\\\[\\\\'m-index\\\\.wxml\\\\'\\\\]::after {
    --tw-content: 'm-index.wxml';
    content: var(--tw-content)
}

",
  "module.js": "(()=>{\\"use strict\\";document.querySelector(\\"#app\\").innerHTML=\\"\\\\n<main class=\\\\\\"after:content-['m-index.js']\\\\\\">\\\\n</main>\\\\n\\"})();",
}
`;

exports[`[Default] > multipleContexts case 1 > errors 1`] = `[]`;

exports[`[Default] > multipleContexts case 1 > warnings 1`] = `[]`;
